### PR TITLE
Php8: update symfony, remove self-update.

### PR DESCRIPTION
**Pull request by ivangrozni**
feature/ldp-1287
feature/ldp-1130

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ To do so, just execute:
  
 ## Updating
 
-Run
+@deprecated: Run
```
Comment (fago) [inline README.md:34]: what's depcreated? let's make sure to update docs as needed

**Review by fago — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ivangrozni) [reply README.md:34]: I've removed self-update command

File: src/Task/Exec.php [hunk 2]
```diff
@@ -96,15 +102,26 @@ protected function ensureCommandRunsViaBash($command) {
...
-    $process->inheritEnvironmentVariables(TRUE);
-
-    return parent::execute($process, $output_callback);
+    /**
+     * {@inheritdoc}
+     */
+  public function run() {
+      $env = [];
+      if ($phapp_env = getenv('PHAPP_ENV')) {
+          $env['PHAPP_ENV'] = $phapp_env;
```
Comment (fago) [inline src/Task/Exec.php:111]: should not be needed. let's remove special-casing of PHAPP_ENV here.

File: src/Task/Exec.php [hunk 3]
```diff
@@ -47,11 +49,15 @@ protected function ensureValidPhappEnvironment() {
     if ($env_command = $this->manifest->getCommand('environment')) {
       $command = trim($env_command) . ' ' . $command ;
     }
-    $process = new Process($this->ensureCommandRunsViaBash($command));
+    $env = [];
+    if ($phapp_env = getenv('PHAPP_ENV')) {
+        $env['PHAPP_ENV'] = $phapp_env;
```
Comment (fago) [inline src/Task/Exec.php:54]: should not be needed. let's remove special-casing of PHAPP_ENV here.

**Review by fago — CHANGES_REQUESTED** [review]

File: src/Task/Exec.php (hunk 3 above)
Comment (ivangrozni) [reply src/Task/Exec.php:54]:
```
$ phapp setup localdev
 [CollectionBuilder]    in task Robo\Collection\CompletionWrapper 

  The phapp environment is undefined. Run "phapp setup $ENVIRONMENT" to initialize it. 
 [CollectionBuilder]  Exit code 1 
 [notice] Stopping on fail. Exiting....
 [error]  Exit Code: 1 
 [error]    in task Robo\Collection\CollectionBuilder 

    in task Robo\Collection\CompletionWrapper 

  The phapp environment is undefined. Run "phapp setup $ENVIRONMENT" to initialize it. 
```

File: src/Task/Exec.php (hunk 2 above)
Comment (ivangrozni) [reply src/Task/Exec.php:111]: This is also needed because otherwise PHAPP_ENV is not in the env

Comment (ivangrozni) [reply src/Task/Exec.php]: @fago I'm not exactly sure how inheritEnvironmentVariables() method used to work, but what we're actually doing is moving php env variables to bash env and it Process doesn't do that automatically.

Comment (fago) [reply src/Task/Exec.php]: I see, we were making sure at least PHAPP_ENV  is set in bash environment. I think we can easily repalce this by simply having
> PHAPP_ENV={{ $ENV }}
as "environment" default value of phapp manifest. So when the loader is not customized, at least the variable is set automatically.

File: README.md [hunk 4]
```diff
@@ -59,9 +54,6 @@ is:
       git:setup-remotes     Configures Git remote repositories.
      init
       init:manifest         Initializes a new phapp.yml for your project.
-     self
-      self:rollback         Rolls back to the previous version after a self-update.
```
Comment (fago) [inline README.md:63]: so let's keep supporting this or similar functionality based upon https://github.com/consolidation/self-update ?

**Review by fago — CHANGES_REQUESTED** [review]

File: src/Task/Exec.php (hunk 3 above)
Comment (ivangrozni) [reply src/Task/Exec.php]: for setup command PHAPP_ENV (in bash) is required so having a check for that should not hurt us. There is this custom override of environment command in setup command (which just clears environment command manifest) but the check is still run.

To further support my findings there is also this method:
```
  protected function initShellEnvironment() {
    // Switch working directory.
    chdir($this->phappManifest->getFile()->getPath());
    // Add the composer bin-dir to the path.
    $path = getenv("PATH");
    putenv("PATH=../vendor/bin/:../bin:$path");
    return $this;
  }
```
This looks like there is another php env var that is supposed to be transfered to the bash variables.

Comment (ivangrozni) [reply src/Task/Exec.php]: agree

**Review by fago — APPROVED** [review]

**Pull request merged**